<?xml version="1.0" ?>
<launch>
  <!-- LAUNCH INTERFACE -->

  <!-- The robot name is required for gazebo to spawn a model. -->
  <arg name="robot_name" default="single_lwr_robot"/>

  <!-- The recommended way of specifying the URDF file is to pass it in
       directly. -->
  <arg name="robot_path" default="$(find single_lwr_robot)/robot"/>
  <arg name="robot_urdf_file" default="$(arg robot_path)/$(arg robot_name).urdf.xacro" />

  <arg name="controllers" default="joint_controllers"/>
  <arg name="t1_limits" default="false"/>

  <arg name="rviz_config" default="$(find lwr_launch)/launch/rviz/rviz_config.rviz"/>
  <arg name="rviz_bringup" default="true"/> <!--If false, do not launch rviz-->

  <arg name="hw_interface_file" default="$(find single_lwr_robot)/config/hw_interface.yaml"/>
  <arg name="controller_config_file" default="$(find single_lwr_robot)/config/controllers.yaml"/>
  <arg name="gazebo_world_file" default="$(find single_lwr_robot)/worlds/simple_environment.world"/>
  <arg name="t1_limits_file" default="$(find single_lwr_robot)/config/t1_joint_limits.yaml"/>
  <arg name="GAZEBO_GUI" default="false"/>

  <!-- ROBOT -->

  <group ns="lwr">
    <param name="robot_description" command="$(find xacro)/xacro.py $(arg robot_urdf_file)"/>
    <param name="use_gui" value="true"/>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>    
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  </group>


  <!-- LAUNCH RVIZ -->

  <node name="lwr_rviz" pkg="rviz" type="rviz" respawn="false" args="-d $(arg rviz_config)"
        output="screen"
        if="$(arg rviz_bringup)"/>

</launch>
